vision/src/autogluon/vision/predictor/predictor.py

```diff
@@ -387,6 +400,16 @@ def _validate_data(self, data):
                               'You may visit `https://auto.gluon.ai/stable/tutorials/image_prediction/dataset.html` ' + \
                               'for details.'
                     raise AttributeError(err_msg)
+            else:
+                raise TypeError(f"Unable to process dataset of type: {type(data)}")
+        elif isinstance(data, _ImageClassification.Dataset):
+            assert 'label' in data.columns
```

Comment:
Could we add `label` as a parameter to init? It could default to `'label'` for now, but I think it would be nice to give users the option to specify the name of the label column as done in Tabular and Text. This would also allow us to transition more naturally into multi-label prediction in future. In general, I think we should avoid hardcoded names.